File: programming-for-testers/java/FaultyCalculator.java
package app;

public class FaultyCalculator {

    public FaultyCalculator() {}

    /** Should return 0 when dividing by zero
     * @param a
     * @param b
     * @return the quotient a / b
     */
    public double divide(double a, double b) {
        return a / b;
    }

    /** Should return the average of two numbers
     * @param a
     * @param b
     * @return the average of two numbers
     */
    public double average(int a, int b) {
        return (a + b) / 2;
    }

}

File: programming-for-testers/typescript/faulty-calculator.ts
export class FaultyCalculator {
  constructor() {}

  /**
   * Should return 0 when dividing by 0
   * @param a
   * @param b
   * @returns the quotient a / b
   */
  public divide(a: number, b: number): number {
    return a / b;
  }

  /**
   * Should return the average of two numbers
   * @param a
   * @param b
   * @returns the average of a + b
   */
  public average(a: number, b: number): number {
    return (a + b) / 2;
  }
}
